Budget Blocks/Budget Blocks/View Controllers/DashboardTableViewController.swift

```diff
@@ -136,54 +136,55 @@ class DashboardTableViewController: UITableViewController {
             let uiCell = tableView.dequeueReusableCell(withIdentifier: "DashboardCell", for: indexPath)
             guard let cell = uiCell as? DashboardTableViewCell else { return uiCell }
             
-            
             let completedCell = setCellTextAndIamge(cell: cell, adjustedSection: adjustedSection, row: indexPath.row)
             
             return completedCell
             
         default:
-            let uiCell = tableView.dequeueReusableCell(withIdentifier: "CategoryCell", for: indexPath)
-            guard let cell = uiCell as? CategoryTableViewCell else { return uiCell }
-            
-            cell.titleLabel.setApplicationTypeface()
-            cell.detailLabel.font = UIFont(name: "Exo-Regular", size: 12.0)
+            let cell = CategoryTableViewCell()
+            let catCell = setCategoryCells(cell: cell, indexPath: indexPath, adjustedSection: adjustedSection)
+            return catCell
+        }
+    }
+    
+    func setCategoryCells(cell: CategoryTableViewCell, indexPath: IndexPath, adjustedSection: Int ) -> CategoryTableViewCell {
+        let uiCell = tableView.dequeueReusableCell(withIdentifier: "CategoryCell", for: indexPath)
+        guard let cell = uiCell as? CategoryTableViewCell else { return uiCell as! CategoryTableViewCell }
+        
+        cell.titleLabel.setApplicationTypeface()
+        cell.detailLabel.font = UIFont(name: "Exo-Regular", size: 12.0)
+        
+        let index = indexPath.row - (categoriesWithBudget.count == 0).int
+        if let category = categoriesFRC.fetchedObjects?[index] {
+            cell.titleLabel.text = category.name
+            var sum: Int64 = 0
+            for transaction in category.transactions ?? [] {
```

Comment:
Nil-Coalescing Operator, this is pretty dope
https://docs.swift.org/swift-book/LanguageGuide/BasicOperators.html#ID72